src/components/Dropdown/Menu.tsx

```diff
@@ -38,29 +45,37 @@ const menuItem = (selected?: boolean) =>
     width('w-full'),
     textAlign('text-left'),
     opacity('disabled:opacity-30'),
-    transitionProperty('transition-colors')
+    transitionProperty('transition-colors'),
+    width({ 'w-full': fitToItemSize }),
+    minWidth({ 'min-w-max': fitToItemSize })
   )
 
 export default function ({
   open,
   options,
   selected,
   onSelect,
+  fitToItemSize,
 }: {
   open: boolean
   options: Option[]
   selected?: string
-  onSelect: (option: Option) => void
+  onSelect?: (option: Option) => void
+  fitToItemSize?: boolean
 }) {
   return (
-    <div className={container(!open)}>
+    <div className={container(!open, fitToItemSize)}>
       <ItemContainer>
         {options.map((option) => (
           <button
             key={option.label}
-            className={menuItem(option.label === selected)}
+            className={menuItem(
+              option.selected || option.label === selected,
+              fitToItemSize
+            )}
             onClick={() => {
-              onSelect(option)
+              if (onSelect) onSelect(option)
+              if (option.href) openLinkInNewTab(option.href)
```

Comment:
We can open the link `onSelect`. I don't think we should check if an option has `href`

Reply:
You're right, I only need to understand how to specify `onSelect` only for one option, not for each like now